packages/obojobo-document-engine/ObojoboDraft/Sections/Assessment/components/rubric/editor.js

```diff
@@ -0,0 +1,405 @@
+	}
+	return json
+}
+
+const plugins = {
+	renderNode(props) {
+		switch (props.node.type) {
+			case MOD_NODE:
+				return <Mod {...props} {...props.attributes} />
+			case MOD_LIST_NODE:
+				return <ModList {...props} {...props.attributes} />
+			case RUBRIC_NODE:
+				return <Node {...props} {...props.attributes} />
+		}
+	},
+	schema: {
+		blocks: {
+			'ObojoboDraft.Sections.Assessment.Rubric': {
+				nodes: [
+					{ match: [{ type: 'Parameter' }], min: 4, max: 4 },
+					{ match: [{ type: MOD_LIST_NODE }], max: 1 }
+				],
+				normalize: (change, error) => {
+					const { node, child, index } = error
+					switch (error.code) {
+						case CHILD_REQUIRED: {
+							let block
+							switch (index) {
+								case 0:
+									block = Block.create(
+										ParameterNode.helpers.oboToSlate({
+											name: 'passingAttemptScore',
+											value: 100 + '',
+											display: 'Passing Score'
+										})
+									)
+									break
+								case 1:
+									block = Block.create(
+										ParameterNode.helpers.oboToSlate({
+											name: 'passedResult',
+											value: 100 + '',
+											display: 'Passed Result'
+										})
+									)
+									break
+								case 2:
+									block = Block.create(
+										ParameterNode.helpers.oboToSlate({
+											name: 'failedResult',
+											value: 0 + '',
+											display: 'Failed Result'
+										})
+									)
+									break
+								case 3:
+									block = Block.create(
+										ParameterNode.helpers.oboToSlate({
+											name: 'unableToPassResult',
+											value: '',
+											display: 'Unable to Pass Result'
+										})
+									)
+									break
+							}
+							return change.insertNodeByKey(node.key, index, block)
+						}
+						case CHILD_TYPE_INVALID: {
+							return change.withoutNormalization(c => {
+								c.removeNodeByKey(child.key)
+								let block
+								switch (index) {
+									case 0:
+										block = Block.create(
+											ParameterNode.helpers.oboToSlate({
+												name: 'passingAttemptScore',
+												value: 100 + '',
+												display: 'Passing Score'
+											})
+										)
+										break
+									case 1:
+										block = Block.create(
+											ParameterNode.helpers.oboToSlate({
+												name: 'passedResult',
+												value: 100 + '',
+												display: 'Passed Result'
+											})
+										)
+										break
+									case 2:
+										block = Block.create(
+											ParameterNode.helpers.oboToSlate({
+												name: 'failedResult',
+												value: 0 + '',
+												display: 'Failed Result'
+											})
+										)
+										break
+									case 3:
+										block = Block.create(
+											ParameterNode.helpers.oboToSlate({
+												name: 'unableToPassResult',
+												value: '',
+												display: 'Unable to Pass Result'
+											})
+										)
+										break
+								}
+								return change.insertNodeByKey(node.key, index, block)
```

Comment:
Above code blocks look similar, is it possible to consolidate them?

Reply:
Not currently, because of the need to wrap the second half in change.withoutNormalizing